platform/qt/src/qmapboxgl.cpp

```diff
@@ -752,66 +742,55 @@ void QMapboxGL::setGestureInProgress(bool progress)
 }
 
 /*!
-    Adds an \a className to the list of active classes. Layers tagged with a certain class
-    will only be active when the class is added.
-
-    This was removed from the \l {https://www.mapbox.com/mapbox-gl-style-spec/#layer-paint.*}
-    {Mapbox style specification} and should no longer be used.
+    Paint classes were removed from the \l {https://www.mapbox.com/mapbox-gl-style-spec/#layer-paint.*}
+    {Mapbox style specification}; this method no longer has any effect.
 
     \deprecated
-    \sa removeClass()
 */
-void QMapboxGL::addClass(const QString &className)
+void QMapboxGL::addClass(const QString &)
```

Comment:
We don't need to maintain binary compatibility on the QMapboxGL interface - we use it internally and expose a separate API on the QtLocation plugin - thus it is safe to assume we can remove all class-handling function implementations from QMapboxGL.